### DISCUSSION TITLE
error when running setup-connections-only

### DISCUSSION THREAD

**Discussion by rboccadoro** [HCL-TECH-SOFTWARE/connections-automation#15] [General]
when I run ansible-playbook -i environments/examples/cnx7/quick_start/connections playbooks/hcl/setup-connections-only.yml" it fails with this error
fatal: [conn7.eldeng.it]: FAILED! => {"changed": false, "msg": "Unable to restart service nfs-server: Failed to restart nfs-server.service: Transaction order is cyclic. See system logs for details. See system logs and 'systemctl status nfs-server.service' for details."}

**Comment by dejo1307** [comment]
The message came from systemd itself, it is impossible to say why NFS couldn't be (re)started without checking the logs the way that error message describes above.

This playbook, in a first step (lines 7 to 11) sets up NFS master and clients. I don't know from the message above if this is related to master (first machine in nfs_masters group) or one of the clients. 

Can you try to figure out first why is NFS failing itself? Then we can try to see how to eventually improve the workflow if needed.

Btw. could take me some time to answer as I am OOO until 4th of January, but will try to help as much as I can.

**Reply by rboccadoro** [reply]
Thanks
no worries, is not urgent at all, I can wait until you return.
Just a question, would it be possible not to use NFS but use local directories ? This is a setup I used for a couple of customers.

**Reply by dejo1307** [reply]
Yes, and I was already pointed out that we should implement it that way (specially for single node installation where NFS is not really needed). 

Btw. if you want to hack it manually, out of my head:
- In https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/main/playbooks/hcl/setup-connections-only.yml comment the lines 7 to 11
- In https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/main/roles/hcl/connections/tasks/main.yml commen the lines 11 and 12

This will fall back to whatever is default where defaults are https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/f4fbae6f34bf42df5c8c99279ed0fd11b3c3761b/roles/hcl/connections/vars/main.yml#L26 and https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/f4fbae6f34bf42df5c8c99279ed0fd11b3c3761b/roles/hcl/connections/vars/main.yml#L31

**Reply by dejo1307** [reply]
I started working on this, my ETA for this is end of this week (it is faster to implement, but counting on testing as well, creating/installing/removing different types of environments).

**Comment by dejo1307** [comment]
This is done for Connections side (and tested), waiting to apply the same for Docs (same use case) and will be merged here as well.
